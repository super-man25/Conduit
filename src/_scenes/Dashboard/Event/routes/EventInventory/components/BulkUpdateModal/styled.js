import styled, { keyframes } from 'styled-components';
import { cssConstants } from '_constants';
import { Box, Text, H4 } from '_components';

export const slideUpAnimation = keyframes`
  0% {
    opacity: 0;
    transform: translate(-50%, -20%);
  }
  100% {
    opacity: 1;
  }
`;

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: 0.3;
  background-color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const ModalContent = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 350px;
  z-index: 9;
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.27);
  animation: ${slideUpAnimation} 200ms ease-out;
  animation-fill-mode: forwards;
`;

export const ModalHeader = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_BLUE};
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
`;

export const ModalBody = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_WHITE};
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
`;

export const Title = styled(H4)`
  margin: 0;
  margin-bottom: 0.25rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const NumberInput = styled.input`
  background-color: ${cssConstants.PRIMARY_WHITE};
  border: 1px solid ${cssConstants.PRIMARY_GRAY};
  border-radius: 3px;
  box-sizing: border-box;
  height: 40px;
  margin: 0;
  padding: 0 0.5rem;
  width: 100%;
`;

export const Label = styled.label`
  display: block;
  margin-bottom: 0.5rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const Field = styled(Box)`
  position: relative;
`;

export const FieldErrorText = styled(Text)`
  position: absolute;
  top: 100%;
`;
