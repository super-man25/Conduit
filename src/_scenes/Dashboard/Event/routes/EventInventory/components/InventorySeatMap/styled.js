import React from 'react';
import { Box, CenteredLoader } from '_components';
import styled from 'styled-components';

export const EventInventoryVerticalRule = styled(Box)`
  background-color: #979797;
  cursor: pointer;
`;

export const SeatIcon = styled.img`
  cursor: pointer;
`;

export const SeatMapLoader = () => (
  <div style={{ position: 'relative', height: '100%', minHeight: 400 }}>
    <CenteredLoader />
  </div>
);
