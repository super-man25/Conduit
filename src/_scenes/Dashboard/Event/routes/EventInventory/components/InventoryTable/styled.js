import styled from 'styled-components';
import { cssConstants } from '_constants';
import { Button, Input, Flex, Box } from '_components';

export const ManualPricingInput = Input.extend`
  box-sizing: border-box;
  margin: 0 0.5rem 0 0;
  padding: 10px 16px;
  max-height: 35px;
  max-width: 80px;
`;

export const ManualPricingButton = Button.extend`
  box-sizing: border-box;
  margin: 0;
  max-width: 80px;
  max-height: 35px;
`;

export const TableHeaderCell = styled.div`
  text-align: ${(props) => props.align || 'left'};
  color: ${cssConstants.PRIMARY_DARKEST_GRAY};
  font-weight: 500;
`;

export const IconContainer = Flex.extend`
  display: inline-flex;
  cursor: pointer;
  position: absolute;
  left: 0;
  top: 50%;
  transform: translateY(-50%);
  opacity: 0.667;
`;

export const PositionedBox = Box.extend`
  position: ${(props) => props.position};
  top: ${(props) => props.top};
  bottom: ${(props) => props.bottom};
  left: ${(props) => props.left};
  right: ${(props) => props.right};
  z-index: ${(props) => props.zIndex};
`;

export const ScaleFilterContainer = Box.extend`
  width: 280px;
  border: 1px solid ${cssConstants.SECONDARY_BLUE};
  background-color: ${cssConstants.PRIMARY_WHITE};
  position: relative;
  box-shadow: 0 2px 5px 0px rgba(0, 0, 0, 0.43);

  &::before {
    content: '';
    width: 8px;
    height: 8px;
    position: absolute;
    top: -5px;
    left: 43px;
    transform: rotate(45deg);
    z-index: 1;
    background-color: ${cssConstants.PRIMARY_WHITE};
    border-top: 1px solid ${cssConstants.SECONDARY_BLUE};
    border-left: 1px solid ${cssConstants.SECONDARY_BLUE};
  }
`;

export const ScaleFilterList = styled.ul`
  margin: 0;
  height: 100%;
  list-style: none;
  padding: 0;
  overflow: auto;
`;

export const ScaleFilterListItem = styled.li`
  height: 35px;
  display: flex;
  padding: 0 1rem;
  align-items: center;
`;

export const Checkbox = styled.input.attrs({ type: 'checkbox' })`
  margin-right: 0.5rem;
`;

export const ScaleFilterButton = styled.button`
  background-color: ${cssConstants.PRIMARY_LIGHT_BLUE};
  border: none;
  border-radius: 2px;
  color: ${cssConstants.PRIMARY_WHITE}
  font-weight: 300;
  height: 25px;
  min-width: 60px;
  padding: 0 1rem;
`;
