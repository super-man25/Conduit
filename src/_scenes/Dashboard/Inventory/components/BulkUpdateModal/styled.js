import styled from 'styled-components';
import { cssConstants } from '_constants';
import { Box, Text, H4, Checkbox } from '_components';

export const ModalHeader = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_BLUE};
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
`;

export const ModalBody = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_WHITE};
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
`;

export const Title = styled(H4)`
  margin: 0;
  margin-bottom: 0.25rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const NumberInput = styled.input`
  background-color: ${cssConstants.PRIMARY_WHITE};
  border: 1px solid ${cssConstants.PRIMARY_GRAY};
  border-radius: 3px;
  height: 40px;
  margin: 0;
  padding: 0 0.5rem;
  width: 100%;
`;

export const Label = styled.label`
  display: block;
  margin-bottom: 0.5rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const Field = styled(Box)`
  position: relative;
`;

export const FieldErrorText = styled(Text)`
  position: absolute;
  top: 100%;
`;

export const StyledCheckbox = styled(Checkbox)`
  margin-top: 15px;
`;
