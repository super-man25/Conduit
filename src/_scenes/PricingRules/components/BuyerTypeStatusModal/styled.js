import styled from 'styled-components';
import { cssConstants } from '_constants';
import { H4, Flex, Input, Text } from '_components';
import { darken } from 'polished';

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: 0.4;
  background-color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const ModalContent = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 500px;
  z-index: 9;
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.27);
`;

export const ModalHeader = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_DARK_BLUE};
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
`;

export const ModalBody = styled.div`
  padding: 1rem 2rem;
  height: 400px;
  overflow-x: hidden;
  overflow-y: scroll;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_DARK_BLUE};
`;

export const ModalFooter = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: ${cssConstants.PRIMARY_DARK_BLUE};
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
`;

export const BuyerTypeOption = Flex.extend`
  padding: 0 2rem;
  min-height: 45px;
  width: 100%;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};

  border-bottom: 1px solid ${cssConstants.PRIMARY_LIGHTER_GRAY};

  :hover {
    background-color: ${darken(0.05, cssConstants.PRIMARY_WHITE)};
  }
`;

export const FlexText = Text.extend`
  flex-basis: ${(props) => props.basis || '10%'};
`;

export const ErrorText = Text.extend`
  text-align: right;
`;

export const Title = H4.extend`
  margin: 0;
  margin-bottom: 2rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const FilterBuyerTypesInput = Input.extend`
  border-radius: 20px;
  height: 35px;
  width: 80px;
`;
