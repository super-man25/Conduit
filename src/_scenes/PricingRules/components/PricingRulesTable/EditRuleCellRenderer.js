// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { Text, Flex } from '_components';
import { SavePricingRuleButton, CancelEditingButton } from './styled';
import { cssConstants } from '_constants';
import { actions } from '_state/priceRule';

type Props = {
  isEditing: boolean,
  rowData: { [string]: any },
  cellData: any,
  isNewRule: boolean,
  isLoading: boolean,
  editingAnyPriceRule: boolean,
  startEditingRule: () => void,
  cancelEditingRule: () => void,
  saveEditedRule: () => void
};

export class EditRuleCellPresenter extends React.Component<Props> {
  startEditingRule = () => {
    const { startEditingRule, editingAnyPriceRule, isLoading } = this.props;
    if (editingAnyPriceRule || isLoading) return;
    startEditingRule();
  };

  cancelEditingRule = () => {
    const { cancelEditingRule } = this.props;
    cancelEditingRule();
  };

  saveEditedRule = () => {
    const { saveEditedRule } = this.props;
    saveEditedRule();
  };

  render() {
    const { editingAnyPriceRule, isEditing, isNewRule, isLoading } = this.props;

    if (!isEditing) {
      return (
        <Flex align="center">
          <Text
            size={14}
            color={
              editingAnyPriceRule || isLoading
                ? cssConstants.PRIMARY_LIGHT_GRAY
                : cssConstants.PRIMARY_LIGHT_BLUE
            }
            onClick={this.startEditingRule}
            style={{ cursor: editingAnyPriceRule ? 'not-allowed' : 'pointer' }}
          >
            EDIT RULE
          </Text>
        </Flex>
      );
    }

    return (
      <Flex justify="space-around">
        <CancelEditingButton
          onClick={this.cancelEditingRule}
          disabled={isLoading}
        >
          Cancel
        </CancelEditingButton>
        <SavePricingRuleButton
          onClick={this.saveEditedRule}
          disabled={isLoading}
        >
          {isNewRule ? 'Create' : 'Save'}
        </SavePricingRuleButton>
      </Flex>
    );
  }
}

const mapStateToProps = (
  { priceRule: { allRows, editingRowId, loading } },
  { rowData }
) => {
  const row = allRows.find((pr) => pr.id === rowData.id);
  return {
    editingAnyPriceRule: editingRowId !== null,
    isNewRule: row.id === 0,
    isLoading: loading
  };
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  startEditingRule: () =>
    dispatch(actions.startEditingPriceRule(ownProps.rowData.id)),
  cancelEditingRule: () =>
    dispatch(actions.cancelEditingPriceRule(ownProps.rowData.id)),
  saveEditedRule: () => dispatch(actions.savePriceRule(ownProps.rowData.id))
});

export const EditRuleCell = connect(
  mapStateToProps,
  mapDispatchToProps
)(EditRuleCellPresenter);
