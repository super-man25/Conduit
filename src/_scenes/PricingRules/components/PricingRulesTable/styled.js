import styled from 'styled-components';
import { cssConstants } from '_constants';
import { PrimaryButton, Input, Box, SecondaryButton } from '_components';

export const EditPricingRuleInput = Input.extend`
  box-sizing: border-box;
  margin: 0 0.5rem 0 0;
  padding: 10px 8px;
  max-height: 35px;
  max-width: 80px;
`;

export const SavePricingRuleButton = PrimaryButton.extend`
  box-sizing: border-box;
  margin: 0;
  min-width: 50px;
  max-width: 75px;
  max-height: 35px;
`;

export const CancelEditingButton = SecondaryButton.extend`
  box-sizing: border-box;
  margin: 0;
  min-width: 50px;
  max-width: 75px;
  max-height: 35px;
  padding: 0 0.75rem;
`;

export const TableHeaderCell = styled.div`
  text-align: ${(props) => props.align || 'left'};
  color: ${cssConstants.PRIMARY_DARKEST_GRAY};
  font-weight: 500;
`;

export const PositionedBox = Box.extend`
  position: ${(props) => props.position};
  top: ${(props) => props.top};
  bottom: ${(props) => props.bottom};
  left: ${(props) => props.left};
  right: ${(props) => props.right};
  z-index: ${(props) => props.zIndex};
`;
export const Checkbox = styled.input.attrs({ type: 'checkbox' })`
  margin-right: 0.5rem;
`;
