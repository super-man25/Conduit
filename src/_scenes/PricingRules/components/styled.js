import styled, { css } from 'styled-components';

import { colors } from '_constants';
import { withClickAway } from '_hoc';
import { Input, Box } from '_components';

export const EditPricingRuleInput = styled(Input)`
  margin: 0 0.5rem 0 0;
  padding: 10px 8px;
  max-height: 35px;
  max-width: 80px;
`;

export const TableHeaderCell = styled.div`
  font-weight: bold;
`;

export const PositionedBox = styled(Box)`
  position: ${(props) => props.position};
  top: ${(props) => props.top};
  bottom: ${(props) => props.bottom};
  left: ${(props) => props.left};
  right: ${(props) => props.right};
  z-index: ${(props) => props.zIndex};
`;
export const Checkbox = styled.input.attrs(() => ({ type: 'checkbox' }))`
  margin-right: 0.5rem;
`;

export const MultiSelectContainer = withClickAway(styled.div`
  position: relative;
  cursor: pointer;
  display: flex;
  justify-content: space-between;
  align-items: center;
`);

export const MultiSelectMenu = styled.div`
  position: absolute;
  width: 250px;
  top: calc(100% + 8px);
  max-height: 350px;
  overflow-y: scroll;
  overflow-x: hidden;
  z-index: 10;
  border: 1px solid ${colors.lightGray};
  box-shadow: 0 20px 20px rgba(0, 0, 0, 0.06);

  transition: 0.1s ease-in-out all;
  opacity: 0;
  transform: translateY(20px);
  visibility: hidden;

  ${(props) =>
    props.show &&
    css`
      opacity: 1;
      transform: translateY(0);
      visibility: visible;
    `}
`;

export const MultiSelectOption = styled.div`
  padding: 12px 16px;
  background-color: ${(props) =>
    props.isActive ? colors.lightGray : colors.white};
  transition: 0.1s ease-in-out all;
  color: ${colors.black};
  white-space: normal;

  :not(:last-child) {
    border-bottom: 1px solid ${colors.lightGray};
  }

  :hover {
    cursor: ${(props) => (props.isActive ? 'default' : 'pointer')};
    background-color: ${colors.lightGray};
  }
`;

export const SplitButtonContainer = styled.div`
  flex-basis: 100%;
  display: flex;
  justify-content: center;
  margin-top: 15px;
`;

export const SplitButtonHalf = styled.span`
  padding: 8px;
  border: 1px solid ${colors.blue};
  display: inline-block;
  min-width: 35%;
  text-align: center;
  cursor: pointer;
  color: ${(props) => (props.isActive ? colors.white : colors.blue)};
  background-color: ${(props) => (props.isActive ? colors.blue : colors.white)};

  :not(:last-child) {
    border-bottom-left-radius: 2px;
    border-top-left-radius: 2px;
    border-right: none;
  }

  :last-child {
    border-bottom-right-radius: 2px;
    border-top-right-radius: 2px;
  }
`;

export const DropdownFilterInputContainer = styled.div`
  background-color: white;
  padding: 10px;
  border-bottom: 1px solid ${colors.lightGray};
`;

export const DropdownFilterInput = styled.input`
  width: 100%;
  padding: 10px 15px;
  border: 1px solid ${colors.lightGray};
  border-radius: 3px;
`;
