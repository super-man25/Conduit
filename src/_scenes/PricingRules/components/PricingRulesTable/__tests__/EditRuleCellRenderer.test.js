import React from 'react';
import { shallow } from 'enzyme';
import { EditRuleCellPresenter } from '../EditRuleCellRenderer';
import { SavePricingRuleButton, CancelEditingButton } from '../styled';
import { Text } from '_components';

describe('<EditRuleCellRenderer />', () => {
  const props = {
    isEditing: false,
    editingAnyPriceRule: false,
    startEditingRule: jest.fn(),
    cancelEditingRule: jest.fn(),
    saveEditedRule: jest.fn()
  };

  it('should render correctly', () => {
    const wrapper = shallow(<EditRuleCellPresenter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger startEditingRule when edit rule is clicked', () => {
    const fn = jest.fn();

    const wrapper = shallow(
      <EditRuleCellPresenter {...props} startEditingRule={fn} />
    );

    wrapper
      .find(Text)
      .at(0)
      .simulate('click');

    expect(fn).toBeCalled();
  });

  it('should trigger cancelEditingRule when cancel button is clicked', () => {
    const fn = jest.fn();

    const wrapper = shallow(
      <EditRuleCellPresenter
        {...props}
        cancelEditingRule={fn}
        isEditing={true}
        editingAnyPriceRule={true}
      />
    );

    wrapper
      .find(CancelEditingButton)
      .at(0)
      .simulate('click');

    expect(fn).toBeCalled();
  });

  it('should trigger saveEditedRule when save button is clicked', () => {
    const fn = jest.fn();

    const wrapper = shallow(
      <EditRuleCellPresenter
        {...props}
        saveEditedRule={fn}
        isEditing={true}
        editingAnyPriceRule={true}
      />
    );

    wrapper
      .find(SavePricingRuleButton)
      .at(0)
      .simulate('click');

    expect(fn).toBeCalled();
  });
});
