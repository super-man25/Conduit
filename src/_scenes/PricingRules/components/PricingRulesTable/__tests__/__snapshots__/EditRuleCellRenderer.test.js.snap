// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditRuleCellRenderer /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditRuleCellPresenter
    cancelEditingRule={[Function]}
    editingAnyPriceRule={false}
    isEditing={false}
    saveEditedRule={[Function]}
    startEditingRule={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "align": "center",
      "children": <styled.p
        color="#38A9DB"
        onClick={[Function]}
        size={14}
        style={
                Object {
                        "cursor": "pointer",
                      }
        }
>
        EDIT RULE
</styled.p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "EDIT RULE",
        "color": "#38A9DB",
        "onClick": [Function],
        "size": 14,
        "style": Object {
          "cursor": "pointer",
        },
      },
      "ref": null,
      "rendered": "EDIT RULE",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "children": <styled.p
          color="#38A9DB"
          onClick={[Function]}
          size={14}
          style={
                    Object {
                              "cursor": "pointer",
                            }
          }
>
          EDIT RULE
</styled.p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "EDIT RULE",
          "color": "#38A9DB",
          "onClick": [Function],
          "size": 14,
          "style": Object {
            "cursor": "pointer",
          },
        },
        "ref": null,
        "rendered": "EDIT RULE",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
